### PR TITLE
add native builds for upx, ucl, zlib

### LINKED ISSUE #988: build UPX natively
See #813

### PR DESCRIPTION
**Pull request by starius**
fix #988

### CODE REVIEW

**Comment by tonytheodore** [comment]
`ucl` and `zlib` don't need `define $(PKG)_BUILD_$(BUILD)` as they're identical to `$(PKG)_BUILD`.

**Comment by starius** [comment]
Unneeded build rules for `ucl` and `zlib` were removed.

**Comment by tonytheodore** [comment]
upx is failing on OSX https://gist.github.com/tonytheodore/48d2f7ddcf1941bfa86e.

It seems to do platform detection within `miniacc.h`, but it's 7,000 lines of `#ifdefs` and very hard to follow.

**Comment by starius** [comment]
Homebrew has UPX: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/upx.rb
Can you compare build processes in Homebrew and in MXE to find differences?
It can be useful to understand what is wrong.

**Comment by tonytheodore** [comment]
I get the same errors compiling by hand using the Homebrew steps and updated lzma. I'll have to look further.

**Comment by tonytheodore** [comment]
`-Werror -Wall` etc. is only set if the compiler is `g++`. 

upx-3.9.1/src/Makefile:29
```makefile
ifeq ($(CXX),)
CXX = g++
endif
ifneq ($(findstring $(firstword $(CXX)),g++),)
USE_GNUC ?= 1
endif
ifeq ($(USE_GNUC),1)
ifeq ($(DEBUG),1)
CXXFLAGS += -O0 -g
else
CXXFLAGS += -O2
endif
CXXFLAGS_WERROR = -Werror
CXXFLAGS += -Wall -W -Wcast-align -Wcast-qual -Wpointer-arith -Wshadow -Wwrite-strings
CXXFLAGS += $(CXXFLAGS_WERROR)
endif
```
Homebrew sets `CXX=clang++` so no warnings are set. https://github.com/LuaAndC/mxe/pull/3 is a conservative way to fix this.

**Comment by tonytheodore** [comment]
Thanks!

Merged: https://github.com/mxe/mxe/compare/31e5701...cc4509c
